packages/core/src/types/config.ts

```diff
@@ -468,6 +468,11 @@ export type BuildCacheOptions = {
    * @default undefined
    */
   cacheDigest?: Array<string | undefined>;
+  /**
+   * An object of arrays of additional code dependencies for the build.
+   * Rspack will use the hash of each of these files to invalidate the persistent cache.
+   */
+  buildDependencies?: Record<string, string[]>;
```

Comment:
How about using string[] to align with Rspack？

Reply:
`string[]` is inconsistent with webpack and I'm worried about whether the rspack cache will change to Object when changing from `experiments` to `stable`.

 https://webpack.js.org/configuration/cache/#cachebuilddependencies

Reply:
We can discuss with @jerrykingxyz first